packages/beacon-node/src/network/network.ts

```diff
@@ -350,4 +358,24 @@ export class Network implements INetwork {
       }
     }
   };
+
+  private async onLightclientFinalityUpdate(finalityUpdate: altair.LightClientFinalityUpdate): Promise<void> {
+    try {
+      // messages SHOULD be broadcasted after one-third of slot has transpired
+      await this.clock.waitForSlot(finalityUpdate.signatureSlot + 1 / 3);
```

Comment:
I don't think this does what you're hoping, this method only works with integer values. Needs to be done a different way.

Reply:
Updated the implementation and also moved to `onSyncAggregate` where the updates are actually emitted. Let me know if that works fine.